File: PLACEMENT_PREPARATION/ZOHO-2025/Navin_kumar_J/Day 1/Four.java
// Find the "celebrity" in a party using a given array of people.
import java.util.*;

class Solution {
    public int celebrity(int mat[][]) {
        int n = mat.length;
        int[] cnt = new int[n];
        HashMap<Integer,Integer> mpp = new HashMap<>();
        for(int i=0;i<n;i++){
            for(int j=0;j<n;j++){
                if(mat[i][j]==1){
                    cnt[j]++;
                    mpp.put(i, mpp.getOrDefault(i, 0) + 1);
                }
            }
        }
        for(int i=0;i<n;i++){
            // System.out.println(cnt[i]);
            if(cnt[i]==n && mpp.get(i)==1){
                return i;
            }
        }
        return -1;
    }
}

public class Four {
    public static void main(String[] args) {
        Solution obj = new Solution();
        int[][] mat =  {{1, 1, 0}, {0, 1, 0}, {0, 1, 1}};
        int ans = obj.celebrity(mat);
        System.out.println(ans);
    }
}
// NAVIN KUMAR J

File: PLACEMENT_PREPARATION/ZOHO-2025/Mariyappan/DAY -1/Four.py
# Find the celebrity in a party using a matrix of people.
def find_celebrity(mat):
    n = len(mat)
    a, b = 0, 1
    while b < n:
        if mat[a][b]:
            a = b
        b += 1
    for i in range(n):
        if i != a and (mat[a][i] or not mat[i][a]):
            return -1
    return a
# Mariyappan
